lib/bookingsync/api/client/conversations.rb

```diff
@@ -0,0 +1,57 @@
+module BookingSync::API
+  class Client
+    module Conversations
+      # List conversations
+      #
+      # Returns all conversations supported in BookingSync.
+      # @param options [Hash] A customizable set of options.
+      # @option options [Array] fields: List of fields to be fetched.
+      # @return [Array<BookingSync::API::Resource>] Array of conversations.
+      #
+      # @example Get the list of conversations for the current account
+      #   conversations = @api.conversations
+      #   conversations.first.subject # => "Question"
+      # @see http://developers.bookingsync.com/reference/endpoints/conversations/#list-conversations
+      def conversations(options = {}, &block)
+        paginate :conversations, options, &block
+      end
+
+      # Get a single conversation
+      #
+      # @param conversation [BookingSync::API::Resource|Integer] Conversation or ID
+      #   of the conversation.
+      # @return [BookingSync::API::Resource]
+      def conversation(conversation)
+        get("conversations/#{conversation}").pop
+      end
+
+      # Create a new conversation
+      #
+      # @param assignee [BookingSync::API::Resource|Integer] User object or ID
+      #   for which assignee conversation will be created.
+      # @param source [BookingSync::API::Resource] Source object or ID
+      #   for which source conversation will be created.
+      # @param options [Hash] Conversation's attributes.
+      # @return [BookingSync::API::Resource] Newly created conversation.
+      def create_conversation(assignee, source, options = {})
+        post(:conversations, conversations: [
+          options.merge(assignee_id: assignee.id, source_id: source.id)
```

Comment:
Why? It should be API consumer's responsibility to provide the proper payload. I'm not even sure if `source` is required so that would force consumer to use null object pattern here.